Equilobe.DailyReport.SL/ReportService.cs

```diff
@@ -23,16 +23,21 @@ public class ReportService : IReportService
         #region IReportService Implementation
         public bool IsDashboardAvailable(DashboardFilter filter)
         {
-            var instance = DataService.GetInstance(filter.InstanceId);
-
-            if (!filter.IsAuthenticated && (string.IsNullOrEmpty(filter.Hash) || instance.Hash != filter.Hash))
+            if (filter.InstanceId == 0 && filter.Hash == "undefined")
```

Comment:
can the hash be undefined?

Reply:
yes, when the user is logged, and there is no hash parameter. It is send as undefined.

Reply:
Why is it marked as undefined? Who marks it that way. It should better be marked null. 
Also, I think the instanceId should be made nullable on the filter, if it can be omitted. And verify if it is null instead of value 0.